packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -276,10 +276,21 @@ impl Experiment {
             env: ExecutionEnvironment::None, // We don't connect to the API
             dyn_payloads: serde_json::Map::from_iter(map_iter),
         };
-        engine_process
+        if let Err(err) = engine_process
             .send(&proto::EngineMsg::Init(init_message))
             .await
-            .wrap_err("Could not send `Init` message")?;
+            .wrap_err("Could not send `Init` message")
+        {
+            // TODO: Wait for threads to finish before starting a forced cleanup
+            warn!("Engine didn't exit gracefully, waiting for threads to finish.");
+            std::thread::sleep(std::time::Duration::from_secs(1));
+
+            engine_process
+                .exit_and_cleanup(experiment_run.base.id)
+                .await
+                .wrap_err("Failed to cleanup after failed start")?;
```

Comment:
We lose `err` if cleanup up fails

Reply:
Wrap the error https://github.com/hashintel/hash/pull/531/commits/49907ed1ead5538083beb12aefe571899f2f647a

Reply:
The two errors are unrelated. Wrapping the error would imply, that the engine process failed to initialize because the cleanup fails.

Reply:
:thinking: I see, is there a way to keep the error? Or should I just ignore it?

Reply:
I'd print a warning if cleanup fails and bail with the original error.

Reply:
Applied https://github.com/hashintel/hash/pull/531/commits/60a670476ad27540287f03a72f55f84233b69bcb